### ISSUE TITLE
 You must not call setTag() on a view Glide is targeting

### ISSUE THREAD

**Issue by fmonsalvo** [DylanVann/react-native-fast-image#221]
Hello,
I'm using the latest version (4.0.14) of fast-image with React-Native 0.53.0.
Today, I was doing some testing on several android devices (Samsung S7 Edge, Nexus 6P, etc) and I'm getting this exception logged in our sentry instance.
You can see the entire stack trace here: https://sentry.keenvil.com/share/issue/a1a259ad75a44809a33943c503bf4229/

Did something change in the android implementation that I should be aware of and are there any code changes I need to make?
Searching through Glide issues I found these ones: [1531](https://github.com/bumptech/glide/issues/1531) and  [370] (https://github.com/bumptech/glide/issues/370) but they both are for Glide 3.7.0 and Fast-Image is on 4.7.1 if I'm not wrong.
Any idea how this can be fixed?
Thanks!

**Comment by DylanVann** [comment]
@fmonsalvo The Glide 4 code hasn't been published to npm yet. It will be in the next major update.

Not sure what could be causing that error. Apparently though you must not call "setTag" on a view glide is targeting, are you possibly doing that somehow? Might be relevant: https://github.com/bumptech/glide/issues/1531

**Comment by PangGua00** [comment]
So do I.  
I used in Flatlist, and when I slided the list over and over again, quickly I got this exception.  
@fmonsalvo could u share the solution? thanks.

**Comment by fmonsalvo** [comment]
Hey @DylanVann this is still happening on latest version (5.0.11). Any chance this will get fixed soon?
Thanks.

**Comment by DavidYangCbre** [comment]
Guys, on my app, whenever I press the back button on Android to close the app or try to log out, this error appears on the android studio logcat. I went and looked up some of the files reported, and the issue for my case seems to be from the FastImageViewManager.java. Below is a snippet of the function reported to be causing the error.

<img width="649" alt="screen shot 2018-12-20 at 4 40 28 pm" src="https://user-images.githubusercontent.com/42848192/50315097-03083680-0477-11e9-9bda-32bab749cd92.png">

The "requestManager.clear(view)" seems to be causing the issue for my app. I tried the brute force approach to see if this was really affecting my app, basically removing the requestManager.clear(view), and the app no longer crashes whenever I press the back button on the home screen or log out.

I don't completely understand the entire library myself or glide since I've only looked at small parts here and there, but this is what I speculate: this function seems to be getting rid of the images, deallocating memory, while glide is still holding onto that view and its images, trying to keep the memory there. With the deallocation, glide is trying to allocate more memory to fill up the lost memories, so the two actions conflict and cause the app to crash.

Not sure if my speculation is near the ballpark or not at all. Can I get someone to tell me the logic behind glide getting rid of the cached images?

Thanks.

**Comment by Damoness** [comment]
![image](https://user-images.githubusercontent.com/5622121/50892737-27ce4900-143a-11e9-8b61-47f70b5db5bb.png)

**Comment by PangGua00** [comment]
In my case, it worked when I removed the property called "testID" in <FastImageView>.

**Comment by Monte9** [comment]
@PangGua00 you are right! Removing `testID` on the `FastImage` component worked for me as well. 

But remember to also add the [`proguard-rules` for android.](https://github.com/DylanVann/react-native-fast-image#proguard)

This issue can be closed imo. Solution was: upgrading to the latest version (v6), adding proguard rules for android & removing `testID` for any `FastImage` component 🎉

**Comment by Sbphillips19** [comment]
I'm had to upgrade everything to react native >0.6 and got everything working for iOS, but haven't figured out android yet.  I am getting an error- `error while updating propety 'source' of a view managed by 'FastImageView'... null ... You must not call setTag() on a view Glide is targeting`

I tried to add the proguard-rules in the READ.me.  I wasn't sure what testID people were talking about, but tried commenting out/ removing the following (from picture below).

Any other suggestions would be great.

```
"react-native": "0.60.5",
"react-native-fast-image": "^7.0.2"
```
<img width="326" alt="Screen Shot 2019-10-09 at 8 57 19 PM" src="https://user-images.githubusercontent.com/25047763/66533778-ba8b3980-ead9-11e9-8abb-70c6032bc4c9.png">
<img width="770" alt="Screen Shot 2019-10-09 at 9 09 03 PM" src="https://user-images.githubusercontent.com/25047763/66533785-be1ec080-ead9-11e9-9cb8-643c8adcc573.png">

**Comment by beisert1** [comment]
We were receiving this crash on Android release builds because we were passing testID in FastImage component. 

Does FastImage not support the testID prop? If not then how do we support jest testing using the FastImage component?

**Comment by Sbphillips19** [comment]
@beisert1 are you passing testID anywhere else.  Still haven't been able to get around this and have been just disabling for android and only using in IOS.

**Comment by okarlsson** [comment]
I had this issue with using react-native-elements `Avatar` component. Seems like it adds the testID under the hood.

Currently just using FastImage for Avatars on iOS:

```
<Avatar
            ImageComponent={Platform.OS === 'ios' ? FastImage : undefined}
            source={{
              uri: picture
            }}
            size={size}
            {...rest}
/>
```

**Comment by vaibhavhrt** [comment]
@okarlsson you can create a custom Image component like this which will work for android and iOS and then pass this component in `ImageComponent` prop of Avatar.

```jsx
import React from 'react';
import FastImage from 'react-native-fast-image';

export const Image = React.forwardRef((props, ref) => {
  const { testID, ...otherProps } = props;

  return <FastImage ref={ref} {...otherProps} />;
});

Image.propTypes = FastImage.propTypes;
Image.priority = FastImage.priority;
Image.resizeMode = FastImage.resizeMode;
Image.cacheControl = FastImage.cacheControl;
Image.preload = FastImage.preload;

export default Image;
```
